src/components/Filters/SchemaSieve.ts

```diff
@@ -1,59 +1,72 @@
 import Ajv from 'ajv';
 import ajvKeywords from 'ajv-keywords';
 import type { JSONSchema7 as JSONSchema } from 'json-schema';
-import cloneDeep from 'lodash/cloneDeep';
-import defaults from 'lodash/defaults';
-import every from 'lodash/every';
-import findKey from 'lodash/findKey';
-import includes from 'lodash/includes';
-import map from 'lodash/map';
-import pickBy from 'lodash/pickBy';
-import reduce from 'lodash/reduce';
-import startsWith from 'lodash/startsWith';
-import trimStart from 'lodash/trimStart';
-import without from 'lodash/without';
+import {
+	generateSchemaFromRefScheme,
+	isJson,
+} from '../../extra/AutoUI/models/helpers';
 import { FilterSignature } from '.';
-import { randomString } from '../../utils';
 import { getDataModel } from '../DataTypes';
+import { randomString } from '../../utils';
+import pickBy from 'lodash/pickBy';
+import findKey from 'lodash/findKey';
+import { isKeyValueObj } from '../DataTypes/object';
+import { getJsonDescription } from '../DataTypes/utils';
 
 const ajv = new Ajv();
 ajvKeywords(ajv, ['regexp', 'formatMaximum', 'formatMinimum']);
 
 export const FULL_TEXT_SLUG = 'full_text_search';
-const DEFAULT_DELIMITER = '___';
+export interface Operator {
+	slug: string;
+	label: string;
+}
+
 interface FullTextFilterItem {
 	key: string;
 	schema: JSONSchema;
-	operator: string;
+	operator: Operator;
 	model: any;
 	searchKey?: string;
 }
 
-export function filter<T>(
-	filters: JSONSchema | JSONSchema[],
-	collection: T[],
-): T[];
-export function filter<T>(
-	filters: JSONSchema | JSONSchema[],
-	collection: Record<string, T>,
-): Record<string, T>;
-export function filter<T>(
-	filters: JSONSchema | JSONSchema[],
-	collection: T[] | Record<string, T>,
-) {
-	// Remove all schemas that may have been compiled already
-	ajv.removeSchema(/^.*$/);
-
-	const validators = Array.isArray(filters)
-		? filters.map((s) => ajv.compile(s))
-		: [ajv.compile(filters)];
+export const getSignatures = (filter: JSONSchema): FilterSignature[] => {
+	const signatures = filter.anyOf!.map((f: JSONSchema) => {
+		return f ? parseFilterDescription(f) : null;
```

Comment:
How does this ternary work if `f` is always `JSONSchema`? What would make it falsy and if it's possible for it to be falsy, shouldn't we add that as a type of `f`?

Reply:
again my bad, this was another piece of code from another try. Thank you for pointing it out :)